### PR TITLE
fixed model not found, added file type messaging

### PR DESCRIPTION
**Pull request by aungkoko**
- fixed model can't be accessed
- fixed for not sending out caption in file type

### CODE REVIEW

File: src/CarroMessenger.php [hunk 1]
```diff
@@ -58,7 +57,7 @@ public function sendWhatsAppTemplateSMSViaMsgBird($data)
             Log::error('WhatsApp template message failed @'. __FUNCTION__.' of '. __CLASS__,
             [$e->getMessage()]);
             
-            event(new MessageWasSent($this->model, new MessageFailedResponse()));
+            event(new MessageWasSent($model, new MessageFailedResponse()));
```
Comment (sithuaung) [inline src/CarroMessenger.php:60]:
```suggestion
            event(new MessageWasSent($model, new MessageFailedResponse()));
```
What if model is null here bro?

Comment (aungkoko) [reply src/CarroMessenger.php:60]: Model won't be null bro..we always pass that model(communication)

File: src/Messaging/WhatsApp/WhatsAppMessageBird.php [hunk 2]
```diff
@@ -78,18 +78,30 @@ public function sendWhatsAppText($to, $message, $reportUrl = null)
...
-        $content->image = [
+
+        $data = [
             'url'       => $imageUrl,
             'caption'   => $caption,
         ];
 
+        $type = 'image';
+
+        if(@exif_imagetype($imageUrl)) {
```
Comment (sithuaung) [inline src/Messaging/WhatsApp/WhatsAppMessageBird.php:89]:
```suggestion
        if(@exif_imagetype($imageUrl)) {
```
why @ is included bro?

Comment (aungkoko) [reply src/Messaging/WhatsApp/WhatsAppMessageBird.php:89]: to prepend error which is caused by using that method bro

**Review by sithuaung — APPROVED** [review]

**Review by yihyang — APPROVED** [review]

**Pull request merged**